saleor/userprofile/views.py

```diff
@@ -8,18 +8,23 @@
 from django.utils.translation import ugettext as _
 
 from .forms import AddressForm
+from .models import User
 
 
 @login_required
 def details(request):
-    ctx = {'addresses': request.user.addresses.all()}
+    user_id = request.user.pk
+    user = User.objects.filter(id=user_id).prefetch_related('addresses')
+    user = user.select_related('default_shipping_address',
+                               'default_billing_address')
+    user = user[0]
```

Comment:
`prefetch_related` is pointless when expecting a single result.

`select_related` saves us one query but it does so making the code very hard to read. It seems the number of queries was already constant before (one for addresses, one for default shipping, one for default billing) so I'm not sure it's worth making the code hard to read to go from three queries to two.